pkg/openstack/rabbitmq.go

```diff
@@ -21,20 +22,84 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-// ReconcileRabbitMQ -
-func ReconcileRabbitMQ(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+type mqStatus int
+
+const (
+	mqFailed   mqStatus = iota
+	mqCreating mqStatus = iota
+	mqReady    mqStatus = iota
+)
+
+// ReconcileRabbitMQs -
+func ReconcileRabbitMQs(
+	ctx context.Context,
+	instance *corev1beta1.OpenStackControlPlane,
+	helper *helper.Helper,
+) (ctrl.Result, error) {
 	if !instance.Spec.Rabbitmq.Enabled {
 		return ctrl.Result{}, nil
 	}
 
+	var failures []string = []string{}
+	var inprogress []string = []string{}
+
+	for name, spec := range instance.Spec.Rabbitmq.Templates {
+		status, err := reconcileRabbitMQ(ctx, instance, helper, name, &spec)
+
+		switch status {
+		case mqFailed:
+			failures = append(failures, fmt.Sprintf("%s(%v)", name, err.Error()))
+		case mqCreating:
+			inprogress = append(inprogress, name)
+		case mqReady:
+		default:
+			return ctrl.Result{}, fmt.Errorf("Invalid mqStatus from reconcileRabbitMQ: %d for RAbbitMQ %s", status, name)
+		}
+	}
+
+	if len(failures) > 0 {
+		errors := strings.Join(failures, ",")
+
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyErrorMessage,
+			errors))
+
+		return ctrl.Result{}, fmt.Errorf(errors)
+
+	} else if len(inprogress) > 0 {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyRunningMessage))
+	} else {
+		instance.Status.Conditions.MarkTrue(
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyCondition,
+			corev1beta1.OpenStackControlPlaneRabbitMQReadyMessage,
+		)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func reconcileRabbitMQ(
+	ctx context.Context,
+	instance *corev1beta1.OpenStackControlPlane,
+	helper *helper.Helper,
+	name string,
+	spec *rabbitmqv1.RabbitmqClusterSpec,
+) (mqStatus, error) {
 	rabbitmq := &rabbitmqv1.RabbitmqCluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "rabbitmq", //FIXME
```

Comment:
the names used here I think can effect k8s native resource names right? (need to double check it) wondering if perhaps we might need to prefix things with 'rabbitmq' + name